### DISCUSSION TITLE
Recommendations for Strongswan client connecting to two remote SAs for HA

### DISCUSSION THREAD

**Discussion by davehouser1** [strongswan/strongswan#682] [Q&A]
**Explanation:**
We are looking for an HA solution for Strongswan host clients connecting to two separate routers running ipsec. 

**Topology:**
```
                  | Juniper VSRX-01 |                                   | Juniper VSRX-02 |
                            |                                                     |
                            |____________________| Cisco Rtr |____________________|
                                                       |
                                                       |
                                                  | Host 01 | 
```

- "Host 01"
    - Will work as a client connecting to the far end network.
    - Will be running Strongswan.
    - Currently uses custom _updown scripts to create VTI interfaces.
    - Interfaces:
        - vti01: created to connect to Juniper-VSRX-01
        - vti02: created to connect to Juniper-VSRX-02
        - ens3: SSH mgnt interface
        - ens4: phyiscal IKE bind interface 
- "Cisco Rtr"
    - This is a L3 router only used to route between the two VSRX's and Host01, it has no other purpose.
- "Juniper VSRX01" and "Juniper VSRX02"
    - These are both running ipsec and bind to specific interfaces that connect to the Cisco Rtr.
    - These systems do not communicate directly with each other in any HA or multi-cluster way. They are seperate entities running separate ipsec services. 
        - Interfaces
            - ge-0/0/0.0 (on both): The physical interface used to bind IKE. Juniper VSRX01 and VSRX02 ge-0/0/0.0 are on separate subnets. 

**Research / trouble shooting**
- Currently able to initialize both tunnels at the same time. However routing does not dynamically update. (Not sure its supposed to?)
    - Test case
        - Updated the custom _updown script for vti01 to send a testing subnet of traffic (10.0.26.0/24) though dev vti01
        - Updated the custom _updown script for vti02 to send a testing subnet of traffic (10.0.26.0/24) though dev vti01 with metric 1
    - Testing performed
        - initialize both tunnels for vti01 --> VSRX01 and vti02 --> VSRX02
        - I check which route the destination host will take, and only one entry shows for vti01. So this wont work:
          ```
          # ip route get 10.0.26.123
          10.0.26.123 dev vti01 src 172.21.0.3 uid 0
                cache
           ```
        - Start a ping to 10.0.26.123, it reaches. In the real world I would test with a dig request as only dns data would go to this ip.
        - I prove that the reroute wont work by simulating an outage on the VSRX01 side, the ping stops, and the higher metric route to vti02 is not used.
        - What is **worse**, is if I reactivate the tunnel on VSRX01, pinging gets no reply, and the tunnel remains down / does not re-establish a connection. 
- Research
    - I read though the [HA documentation recommendations](https://wiki.strongswan.org/projects/strongswan/wiki/highavailability). The document Seems to be laid out as a way to configure Strongswan to take the role of the Juniper VSRX's in the above topology. If one strongswan system goes down, the other one will kick in. I do not believe I am looking for this. 
- Workaround
    - I scripted a failover system that will recognize if a tunnel is down based of icmp heart beats it creates. If it recognizes one system is down it will run though a variety of configurations to shift all traffic to the other tunnel. This is not elegant in anyway, also there is a short time period of outage while the traffic is shifted to the other tunnel, its short, but not good. Can Strongswan utilize this functionality natively? 

**The Want**
- Need a way so the client (Host01) is always connected to at least one router via ipsec. 
- The Juniper VSRX routers will be support multiple remote SA's (Like Host01).
- In the end only DNS traffic will traverse the ipsec tunnel. The destination hosts for the DNS traffic will be to anywhere. 
- SSH mgnt traffic must still be able to reach the Host01 interface ens3
    - The way I have been accomplishing this thus far is to send all traffic (0.0.0.0/0) via the tunnel, and adding static routes out other interfaces (ssh mgnt) to avoid use of the tunnel.
- Would like both tunnels active, if possible, but if only one can be active at time so be it.
- Both tunnels must be able to rekey locally or remotely automatically. Tunnels must be able to rebuild on their own once the connections are restored. 

**Thoughts**
- I believe I am going about this wrong, but its not clear to me how to do this correctly. Am I missing specific settings somewhere? or using settings that are causing problems? Or is my whole design flawed and I rethink the way I am doing things? 

**/etc/strongswan/swanctl/swanctl.conf**
```
connections {
    vsrx-01b {
        version = 2
        local_addrs = 2.2.1.2
        remote_addrs = 3.3.0.2
        proposals = aes256-sha256-ecp384
        local {
            auth = psk
        }
        remote {
            auth = psk
        }
        children {
            vsrx-01b-child {
                local_ts = 172.21.0.0/29
                remote_ts = 0.0.0.0/0
                mark_in = 41
                mark_out = 41
                esp_proposals = aes256-sha256-ecp384
                updown = /opt/_updown_vti01
                start_action = start
                close_action = restart
            }
        }
    }
    vsrx-02b {
        version = 2
        local_addrs = 2.2.1.2
        remote_addrs = 3.3.1.2
        proposals = aes256-sha256-ecp384
        local {
            auth = psk
        }
        remote {
            auth = psk
        }
        children {
            vsrx-02b-child {
                local_ts = 172.22.0.0/29
                remote_ts = 0.0.0.0/0
                mark_in = 42
                mark_out = 42
                esp_proposals = aes256-sha256-ecp384
                updown = /opt/_updown_vti02
                start_action = start
                close_action = restart
            }
        }
    }
}
secrets {
    ike-me {
        secret = "PaSsW0rd"
        id-to-vsrx-01b = 2.2.1.2
    }
    ike-vsrx {
        secret = "PaSsW0rd"
        id-to-vsrx-01b = 3.3.0.2
    }
}
include conf.d/*.conf
```

**/opt/_updown_vti01**
```
#!/bin/bash

set -o nounset
set -o errexit

VTI_IF="vti01"

case "${PLUTO_VERB}" in
    up-client)
        ip tunnel add $VTI_IF local 2.2.1.2 remote 3.3.0.2 mode vti key 41
        ip link set $VTI_IF up
        ip addr add  172.21.0.3 dev $VTI_IF
        ip route add 172.21.0.0/29 dev $VTI_IF
        ip route add 10.0.26.0/24 dev $VTI_IF
        sysctl -w "net.ipv4.conf.$VTI_IF.disable_policy=1"
        sysctl -w "net.ipv4.conf.$VTI_IF.rp_filter=0"
        sysctl -w "net.ipv4.conf.$VTI_IF.forwarding=1"
        ;;
    down-client)
        ip tunnel del $VTI_IF
        ;;
esac
```

**/opt/_updown_vti02**
```
!/bin/bash

set -o nounset
set -o errexit

VTI_IF="vti02"

case "${PLUTO_VERB}" in
    up-client)
        ip tunnel add $VTI_IF local 2.2.1.2 remote 3.3.1.2 mode vti key 42
        ip link set $VTI_IF up
        ip addr add  172.22.0.3 dev $VTI_IF
        ip route add 172.22.0.0/29 dev $VTI_IF
        ip route add 10.0.26.0/24 dev $VTI_IF metric 1
        sysctl -w "net.ipv4.conf.$VTI_IF.disable_policy=1"
        sysctl -w "net.ipv4.conf.$VTI_IF.rp_filter=0"
        sysctl -w "net.ipv4.conf.$VTI_IF.forwarding=1"
        ;;
    down-client)
        ip tunnel del $VTI_IF
        ;;
esac
```

**/etc/strongswan/strongswan.conf**
```
charon {
    load_modular = yes 
    make_before_break = yes 
    install_routes = no
    plugins {
        include strongswan.d/charon/*.conf
    }   
    filelog {
        charon{
            path = /var/log/charon.log
            default = 4 
        }
    }   
}

include strongswan.d/*.conf
```

**`ip -br a` and `route -n` after both tunnels are initalized**
```
# ip -br a
lo               UNKNOWN        127.0.0.1/8 ::1/128
ens3             UP             192.168.0.15/24 fe80::e74:fcff:fe28:cb00/64
ens4             UP             2.2.1.2/30 fe80::e74:fcff:fe28:cb01/64
virbr0           DOWN           192.168.122.1/24
virbr0-nic       DOWN
ip_vti0@NONE     DOWN
vti01@NONE       UNKNOWN        172.21.0.3/32 fe80::200:5efe:202:102/64
vti02@NONE       UNKNOWN        172.22.0.3/32 fe80::200:5efe:202:102/64

# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
2.2.0.1         0.0.0.0         255.255.255.255 UH    101    0        0 ens4
2.2.1.0         0.0.0.0         255.255.255.252 U     101    0        0 ens4
3.3.0.0         2.2.0.1         255.255.255.252 UG    101    0        0 ens4
3.3.1.0         2.2.1.1         255.255.255.252 UG    101    0        0 ens4
10.0.26.0       0.0.0.0         255.255.255.0   U     0      0        0 vti01
10.0.26.0       0.0.0.0         255.255.255.0   U     1      0        0 vti02
10.212.134.0    192.168.0.1     255.255.255.0   UG    100    0        0 ens3
172.21.0.0      0.0.0.0         255.255.255.248 U     0      0        0 vti01
172.22.0.0      0.0.0.0         255.255.255.248 U     0      0        0 vti02
192.168.0.0     0.0.0.0         255.255.255.0   U     100    0        0 ens3
192.168.122.0   0.0.0.0         255.255.255.0   U     0      0        0 virbr0
```

**watching `swanctl --list-sas` while both are connected**
```
Every 1.0s: swanctl --list-sas                                                                                                                                                                                                       gst-02b: Fri Oct 22 20:28:54 2021

vsrx-02b: #2, ESTABLISHED, IKEv2, 9e01e0aada255f60_i* bc63585a99439880_r
  local  '2.2.1.2' @ 2.2.1.2[4500]
  remote '3.3.1.2' @ 3.3.1.2[4500]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_384
  established 49s ago, rekeying in 13146s
  vsrx-02b-child: #2, reqid 2, INSTALLED, TUNNEL, ESP:AES_CBC-256/HMAC_SHA2_256_128
    installed 49s ago, rekeying in 3374s, expires in 3911s
    in  c489eca7 (0x0000002a),      0 bytes,     0 packets
    out 91d13151 (0x0000002a),      0 bytes,     0 packets
    local  172.22.0.0/29
    remote 0.0.0.0/0
vsrx-01b: #1, ESTABLISHED, IKEv2, b8ea3f9165403fb6_i* 4e3564e28072c4d2_r
  local  '2.2.1.2' @ 2.2.1.2[4500]
  remote '3.3.0.2' @ 3.3.0.2[4500]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_384
  established 56s ago, rekeying in 14326s
  vsrx-01b-child: #1, reqid 1, INSTALLED, TUNNEL, ESP:AES_CBC-256/HMAC_SHA2_256_128
    installed 56s ago, rekeying in 3247s, expires in 3904s
    in  c3a3377c (0x00000029),      0 bytes,     0 packets
    out 5411809f (0x00000029),      0 bytes,     0 packets
    local  172.21.0.0/29
    remote 0.0.0.0/0
```

**Comment by Thermi** [comment]
Don't automatically create VTIs via updown script if you can help it. It is only possible do to it well via workarounds.
Try to use XFRM interfaces that are statically created before strongSwan starts and configure your tunnels to use them with if_id_in and if_id_out. Check if your linux kernel version supports that.

Please provide the ip and route information exactly as shown in #196.

You can implement the behaviour with preferably dynamic routing.

**Reply by davehouser1** [reply]
> You can implement the behaviour with preferably dynamic routing.

Great to hear, how can I go about accomplishing this? I read the HA / Failover section [here ](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#High-Availability-and-Failover-configurations). It says its possible, but I cant find any examples for failover. How do I setup dynamic routing tables for failover? I see [older posts](https://wiki.strongswan.org/issues/2393) about how this is not possible, is that true? 

Here is the ip and route info in the correct format after both tunnels are up. 

```
# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 0c:74:fc:28:cb:00 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.15/24 brd 192.168.0.255 scope global noprefixroute ens3
       valid_lft forever preferred_lft forever
    inet6 fe80::e74:fcff:fe28:cb00/64 scope link
       valid_lft forever preferred_lft forever
3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 0c:74:fc:28:cb:01 brd ff:ff:ff:ff:ff:ff
    inet 2.2.1.2/30 brd 2.2.1.3 scope global noprefixroute ens4
       valid_lft forever preferred_lft forever
    inet6 fe80::e74:fcff:fe28:cb01/64 scope link
       valid_lft forever preferred_lft forever
4: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 52:54:00:2b:5d:cd brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
       valid_lft forever preferred_lft forever
5: virbr0-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc fq_codel master virbr0 state DOWN group default qlen 1000
    link/ether 52:54:00:2b:5d:cd brd ff:ff:ff:ff:ff:ff
6: ip_vti0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
222: vti02@NONE: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1480 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ipip 2.2.1.2 peer 3.3.1.2
    inet 172.22.0.3/32 scope global vti02
       valid_lft forever preferred_lft forever
    inet6 fe80::200:5efe:202:102/64 scope link
       valid_lft forever preferred_lft forever
223: vti01@NONE: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1480 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ipip 2.2.1.2 peer 3.3.0.2
    inet 172.21.0.3/32 scope global vti01
       valid_lft forever preferred_lft forever
    inet6 fe80::200:5efe:202:102/64 scope link
       valid_lft forever preferred_lft forever
       
# ip route show table all
2.2.0.1 dev ens4 proto static scope link metric 101
2.2.1.0/30 dev ens4 proto kernel scope link src 2.2.1.2 metric 101
3.3.0.0/30 via 2.2.0.1 dev ens4 proto static metric 101
3.3.1.0/30 via 2.2.1.1 dev ens4 proto static metric 101
10.0.26.0/24 dev vti01 scope link
10.0.26.0/24 dev vti02 scope link metric 1
10.212.134.0/24 via 192.168.0.1 dev ens3 proto static metric 100
172.21.0.0/29 dev vti01 scope link
172.22.0.0/29 dev vti02 scope link
192.168.0.0/24 dev ens3 proto kernel scope link src 192.168.0.15 metric 100
192.168.122.0/24 dev virbr0 proto kernel scope link src 192.168.122.1 linkdown
broadcast 2.2.1.0 dev ens4 table local proto kernel scope link src 2.2.1.2
local 2.2.1.2 dev ens4 table local proto kernel scope host src 2.2.1.2
broadcast 2.2.1.3 dev ens4 table local proto kernel scope link src 2.2.1.2
broadcast 127.0.0.0 dev lo table local proto kernel scope link src 127.0.0.1
local 127.0.0.0/8 dev lo table local proto kernel scope host src 127.0.0.1
local 127.0.0.1 dev lo table local proto kernel scope host src 127.0.0.1
broadcast 127.255.255.255 dev lo table local proto kernel scope link src 127.0.0.1
local 172.21.0.3 dev vti01 table local proto kernel scope host src 172.21.0.3
local 172.22.0.3 dev vti02 table local proto kernel scope host src 172.22.0.3
broadcast 192.168.0.0 dev ens3 table local proto kernel scope link src 192.168.0.15
local 192.168.0.15 dev ens3 table local proto kernel scope host src 192.168.0.15
broadcast 192.168.0.255 dev ens3 table local proto kernel scope link src 192.168.0.15
broadcast 192.168.122.0 dev virbr0 table local proto kernel scope link src 192.168.122.1 linkdown
local 192.168.122.1 dev virbr0 table local proto kernel scope host src 192.168.122.1
broadcast 192.168.122.255 dev virbr0 table local proto kernel scope link src 192.168.122.1 linkdown
::1 dev lo proto kernel metric 256 pref medium
fe80::/64 dev ens3 proto kernel metric 256 pref medium
fe80::/64 dev ens4 proto kernel metric 256 pref medium
fe80::/64 dev vti02 proto kernel metric 256 pref medium
fe80::/64 dev vti01 proto kernel metric 256 pref medium
local ::1 dev lo table local proto kernel metric 0 pref medium
local fe80::200:5efe:202:102 dev vti02 table local proto kernel metric 0 pref medium
local fe80::200:5efe:202:102 dev vti01 table local proto kernel metric 0 pref medium
local fe80::e74:fcff:fe28:cb00 dev ens3 table local proto kernel metric 0 pref medium
local fe80::e74:fcff:fe28:cb01 dev ens4 table local proto kernel metric 0 pref medium
ff00::/8 dev ens3 table local metric 256 pref medium
ff00::/8 dev ens4 table local metric 256 pref medium
ff00::/8 dev vti02 table local metric 256 pref medium
ff00::/8 dev vti01 table local metric 256 pref medium
```

**Comment by davehouser1** [comment]
Bumping this again, can anyone help? I read though [this link](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#High-Availability-and-Failover-configurations), this section reads:

_"However, with route based tunnels that are built using VTIs and with a dynamic routing daemon, such a configuration should be possible between one strongSwan installation and two redundant remote gateways"_

That is exactly what I have except unit with dynamic routing (yet). Any recommendations or configuration examples for getting failover between two vti's to work? Any recommendations for using a dynamic routing daemon (Like Quagga)?

**Comment by Thermi** [comment]
Hi,

We've made that work with xfrm interfaces and frr's bgp daemon. You need to set the multicast flag on the interfaces manually afair, then it works.

Kind regards
Noel

Am 12. November 2021 13:47:40 UTC schrieb davehouser1 ***@***.***>:
>Bumping this again, can anyone help? I read though [this link](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#High-Availability-and-Failover-configurations), this section reads:
>
>_"However, with route based tunnels that are built using VTIs and with a dynamic routing daemon, such a configuration should be possible between one strongSwan installation and two redundant remote gateways"_
>
>That is exactly what I have except unit with dynamic routing (yet). Any recommendations or configuration examples for getting failover between two vti's to work? Any recommendations for using a dynamic routing daemon (Like Quagga)?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/682#discussioncomment-1631939
-- 
Sent from mobile

**Reply by davehouser1** [reply]
To confirm, VTI interfaces wont cut it? I need to use XFRM interfaces? Also are there any notes you can share on how you accomplished this?

**Reply by Thermi** [reply]
bgpd of frr requires that the multicast flag is set on the device and there's an actual network on it. I set it up with XFRMis once, and recently saw a config with VTIs that worked (I configured it for a customer).
You can definitely do it. It's just much easier with XFRMis.

**Reply by davehouser1** [reply]
Thanks, yes I upgraded our kernel to newer version, an deployed XFMRi's. I also installed frr, and enabled the bgp daemon. 
Did you do iBGP or eBGP to each SA? Curious what was the far end SA devices? (Cisco, Juniper, other FRR linux boxes, etc.)?
If you could share your frr / far end router SA's configs that would be very helpful. I am getting closer with this but not there yet.

**Reply by Thermi** [reply]
Sure, attached is an anonymized version. It's used to talk to another peer over a redundant VPN connection to a CISCO peer within GCP.

Are you having trouble associating the XFRMis or something else?

[example-frr.txt](https://github.com/strongswan/strongswan/files/7533122/example-frr.txt)

**Reply by davehouser1** [reply]
Thank you. Currently I am trying to learn FRR a little more. I messed with it with the SONiC NOS, but not super familiar with how it works. So I am trying to get basic L3  bgp working first. The strongswan host  currently connects to juniper vsrx's and Cicso IOS L3 nodes via ibgp. Once I get route redistribution working then I will move on to getting bgp to work over IPsec. I will probably return to this thread if I run into more roadblocks. I know this thread is not a FRR or bgp discussion board so I appreciate what your sharing.

**Reply by davehouser1** [reply]
@Thermi ok so I tested bgp between my host running frr and the two vsrx's (this worked). Be aware I did not do this over IPsec.
Now I am trying to configure bgp over IPsec. I changed my vsrx's to only connect to the peer over the IPsec interfaces, and adjust frr to use the IPsec interfaces as well of the neighbors. However when I enable IPsec, I see no bgp packets ingress or egress though the tunnel. I know I can send icmp via the xfrmi's to the IPsec interfaces on the vsrx's, but still no bgp.

I am willing to share configs and details, but was wondering if you encountered the same problem at all? Firewalld is disabled, I am allowing all subnets for right_ts and left_ts, I made sure all applications and ips are accepted via the vsrx zones. Can't figure out why bgp packets won't traverse the tunnel. 

You mentioned enabling multicast, can you go into a little more details about that? Was that on the Linux kernel? sysctl? In frr? Where did you do this and how can I check if I am doing the same correctly?

**Comment by Thermi** [comment]
Hi, you need to get debug logging for bgpd going and set the Multicast flag on the xfrmis. But get debugging going first. It will tell you what the daemon does and why it does it.

Am 16. November 2021 18:00:14 UTC schrieb davehouser1 ***@***.***>:
***@***.*** ok so I tested bgp between my host running frr and the two vsrx's (this worked). Be aware I did not do this over IPsec.
>Now I am trying to configure bgp over IPsec. I changed my vsrx's to only connect to the peer over the IPsec interfaces, and adjust frr to use the IPsec interfaces as well of the neighbors. However when I enable IPsec, I see no bgp packets ingress or egress though the tunnel. I know I can send icmp via the xfrmi's to the IPsec interfaces on the vsrx's, but still no bgp.
>
>I am willing to share configs and details, but was wondering if you encountered the same problem at all? Firewalld is disabled, I am allowing all subnets for right_ts and left_ts, I made sure all applications and ips are accepted via the vsrx zones. Can't figure out why bgp packets won't traverse the tunnel. 
>
>You mentioned enabling multicast, can you go into a little more details about that? Was that on the Linux kernel? sysctl? In frr? Where did you do this and how can I check if I am doing the same correctly?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/682#discussioncomment-1653220
-- 
Sent from mobile

**Comment by Thermi** [comment]
afair `ip link set x multicast on`

Am 16. November 2021 18:00:14 UTC schrieb davehouser1 ***@***.***>:
***@***.*** ok so I tested bgp between my host running frr and the two vsrx's (this worked). Be aware I did not do this over IPsec.
>Now I am trying to configure bgp over IPsec. I changed my vsrx's to only connect to the peer over the IPsec interfaces, and adjust frr to use the IPsec interfaces as well of the neighbors. However when I enable IPsec, I see no bgp packets ingress or egress though the tunnel. I know I can send icmp via the xfrmi's to the IPsec interfaces on the vsrx's, but still no bgp.
>
>I am willing to share configs and details, but was wondering if you encountered the same problem at all? Firewalld is disabled, I am allowing all subnets for right_ts and left_ts, I made sure all applications and ips are accepted via the vsrx zones. Can't figure out why bgp packets won't traverse the tunnel. 
>
>You mentioned enabling multicast, can you go into a little more details about that? Was that on the Linux kernel? sysctl? In frr? Where did you do this and how can I check if I am doing the same correctly?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/682#discussioncomment-1653220
-- 
Sent from mobile

**Reply by davehouser1** [reply]
I was able to enable debugging (I think), `log file /var/log/frr/bgpd.log debugging`.
This is the output for FRR

```
2021/11/17 01:19:23 BGP: [JFMSW-YMBC7] 172.21.0.1 [FSM] Timer (connect timer expire)
2021/11/17 01:19:23 BGP: [ZWCSR-M7FG9] 172.21.0.1 [FSM] ConnectRetry_timer_expired (Active->Connect), fd -1
2021/11/17 01:19:23 BGP: [J47J0-K06GG] Found existing bnc 172.21.0.1/32(VRF default) flags 0xe ifindex 0 #paths 0 peer 0x7fb944a75010
2021/11/17 01:19:23 BGP: [T72VK-55DVG] 172.21.0.1 [FSM] Waiting for NHT
2021/11/17 01:19:23 BGP: [T91AW-FGMHW] bgp_fsm_change_status : vrf default(0), Status: Connect established_peers 0
2021/11/17 01:19:23 BGP: [ZQHFG-DQGX1] 172.21.0.1 went from Active to Connect
2021/11/17 01:19:23 BGP: [ZWCSR-M7FG9] 172.21.0.1 [FSM] TCP_connection_open_failed (Connect->Active), fd -1
2021/11/17 01:19:24 BGP: [T91AW-FGMHW] bgp_fsm_change_status : vrf default(0), Status: Active established_peers 0
2021/11/17 01:19:24 BGP: [ZQHFG-DQGX1] 172.21.0.1 went from Connect to Active
2021/11/17 01:20:09 BGP: [JFMSW-YMBC7] 172.22.0.1 [FSM] Timer (connect timer expire)
2021/11/17 01:20:09 BGP: [ZWCSR-M7FG9] 172.22.0.1 [FSM] ConnectRetry_timer_expired (Active->Connect), fd -1
2021/11/17 01:20:09 BGP: [J47J0-K06GG] Found existing bnc 172.22.0.1/32(VRF default) flags 0xe ifindex 0 #paths 0 peer 0x7fb94491f010
2021/11/17 01:20:09 BGP: [T72VK-55DVG] 172.22.0.1 [FSM] Waiting for NHT
2021/11/17 01:20:09 BGP: [T91AW-FGMHW] bgp_fsm_change_status : vrf default(0), Status: Connect established_peers 0
2021/11/17 01:20:09 BGP: [ZQHFG-DQGX1] 172.22.0.1 went from Active to Connect
2021/11/17 01:20:09 BGP: [ZWCSR-M7FG9] 172.22.0.1 [FSM] TCP_connection_open_failed (Connect->Active), fd -1
2021/11/17 01:20:09 BGP: [T91AW-FGMHW] bgp_fsm_change_status : vrf default(0), Status: Active established_peers 0
2021/11/17 01:20:09 BGP: [ZQHFG-DQGX1] 172.22.0.1 went from Connect to Active
```

This appears to be the same problem on the VSRX's sides, bgp is just not able to send any packets.
Before I found out that my VSRX was not allowing bgp on its security zones, I enabled this for all services and all protocols on all zones. Still no matter what, no bgp packets refuse to traverse this tunnel. I have no idea why. 

Also I cant quite figure out how to enable multicast for a xfrmi, how did you accomplish this? Also not sure how to tell if an xfmri has multicast enabled or not. Lastly why does multicast need to be enabled? Isn't BGP TCP? I thought multicast was for UDP, why does this whole architecture require multicast?

**Reply by davehouser1** [reply]
Just an update, I go this working with ebgp! Problem was firewall rules + multihop. Didnt have to manually enable multicast, so not sure if the design is flawed because of this or if multicast is already enabled.
I plan to implement via igp(ospf) before sharing my work. These connections for our architecture are not edge, so ebgp implementation was just a POC for us. 
Thanks again @Thermi for the help!

**Reply by Thermi** [reply]
Great! You are very welcome. Would you share your configurations with us, please?
